virtool/tasks/runner.py

```diff
@@ -48,9 +66,14 @@ async def run_task(self, task_id: int):
         """
         task: Task = await get_row_by_id(self.app["pg"], Task, task_id)
 
-        logging.info(f"Starting task: %s %s", task.id, task.type)
+        logging.info("Starting task id=%s name=%s", task.id, task.type)
 
         for cls in BaseTask.__subclasses__():
             if task.type == cls.name:
                 current_task = await cls.from_task_id(self.app["data"], task.id)
-                await asyncio.get_event_loop().create_task(current_task.run())
+
+                self._current_task = asyncio.create_task(
+                    current_task.run()
+                )
+
+                await asyncio.shield(self._current_task)
```

Comment:
This shield is required since we need to protect the underlying task from being canceled when the asycio canceled error is propagated